thinkhazard/static/js/report.js

```diff
@@ -129,13 +129,31 @@
       fillStyle.setColor(fillColor);
       return styles;
     };
-    var interaction = new ol.interaction.Select({
+    var clickInteraction = new ol.interaction.Select({
       layers: [layer],
-      condition: ol.events.condition.pointerMove,
-      style: styleFn
+      condition: ol.events.condition.click
     });
-    map.addInteraction(interaction);
-    return interaction;
+    var interactions = [
+      new ol.interaction.Select({
+        layers: [layer],
+        condition: ol.events.condition.pointerMove,
+        style: styleFn
+      }),
+      clickInteraction
+    ];
+    clickInteraction.on('select', function(e) {
+      var feature = e.selected[0];
+      if (!feature) {
+        return;
```

Comment:
I disagree with this way of thinking :) To me either you know that the `selected` array always has elements and you don't use a test, or you know that there are cases where the `selected` array is empty and you use a test. But you should not add a test because you don't know. I'd argue that this is an anti-pattern. 

And what about `e.selected`? You access its first element without testing that it's not `undefined`? Do you know that it cannot be `undefined`?